splash/auth/oauth_resources.py

```diff
@@ -0,0 +1,47 @@
+from bson.json_util import dumps
+import jsonschema
+import json
+import os
+from flask import current_app
+from flask import request
+from flask_restful import Resource
+from google.oauth2 import id_token
+from google.auth.transport import requests
+
+dirname = os.path.dirname(__file__)
+auth_schema_file = open(os.path.join(dirname, "auth_schema.json"))
+AUTH_SCHEMA = json.load(auth_schema_file)
+auth_schema_file.close()
+
+
+CLIENT_ID = os.environ.get("GOOGLE_CLIENT_ID") + ".apps.googleusercontent.com" #TODO: Temporary solution, we will need to have a local config file
+#TODO: Find out how to handle errors for this endpoint
+
+class OAuthResource(Resource):
+    def post(self):
+        #TODO: Verify the schema before blindly assigning this to a variable
+        validator = jsonschema.Draft7Validator(AUTH_SCHEMA)
+        json = request.get_json(force=True)
```

Comment:
You have imported a package called json before this. Again, probably copied from other code that needs to be changed, but you should change this variable name so that there is no conflict.

Also, can you consider what happens if there is no json payload?

Reply:
Ah yes, I'll make sure to fix the naming conflict. Also If there is no JSON payload then `request.get_json` will return `None` I believe. The validator will then throw a validation error if it is passed `None`. Although, I will ensure to test this behavior in the unit tests.

Reply:
You could save some processing by not trying to validate if there is no json. Also, be weary of the call after validate.

Reply:
Which call should I be weary of and how?

Reply:
I was wrong here also. If their is an error in parsing the JSON then the function will raise a BadRequest error. So I think we can keep the code here as is.